pathfinder/pathfinding_service.py

```diff
@@ -181,19 +182,32 @@ def on_fee_info_message(self, fee_info: FeeInfo):
             log.debug('Received FeeInfo message for token network {}'.format(
                 token_network.address
             ))
-
-            fee_info_signer = Address(
-                eth_verify(decode_hex(fee_info.signature), fee_info.serialize_bin())
-            )
             # TODO: check chain id
 
             token_network.update_fee(
                 fee_info.channel_identifier,
-                fee_info_signer,
+                Address(fee_info.signer),
                 fee_info.nonce,
                 fee_info.percentage_fee
             )
 
+    def on_balance_proof_message(self, balance_proof: BalanceProof):
+        token_network = self._get_token_network(balance_proof.token_network_address)
+
+        if token_network:
+            log.debug('Received BalanceProof message for token network {}'.format(
+                token_network.address
+            ))
+            # TODO: check chain id
+
+            token_network.update_balance(
+                balance_proof.channel_identifier,
+                Address(balance_proof.signer),
+                balance_proof.nonce,
+                balance_proof.transferred_amount,
+                balance_proof.locked_amount,
+            )
+
     def handle_token_network_created(self, event):
         token_network_address = event['args']['token_network_address']
         assert is_checksum_address(token_network_address)
```

Comment:
I'd suggest to add a test for this into integration_test, in order not to have any untested methods in the code base.